pkg/csi/service/cns/nodes.go

```diff
@@ -48,15 +48,23 @@ func (nodes *Nodes) Initialize() error {
 	}
 	nodes.cnsNodeManager.SetKubernetesClient(k8sclient)
 	nodes.informMgr = k8s.NewInformer(k8sclient)
-	nodes.informMgr.AddNodeListener(nodes.nodeAdd, nil, nodes.nodeDelete)
+	nodes.informMgr.AddNodeListener(nodes.nodeAdd, nodes.nodeUpdate, nodes.nodeDelete)
 	nodes.informMgr.Listen()
 	return nil
 }
 
 func (nodes *Nodes) nodeAdd(obj interface{}) {
+	nodes.nodeRegister(obj)
+}
+
+func (nodes *Nodes) nodeUpdate(oldObj interface{}, newObj interface{}) {
```

Comment:
Have you done any testing to validate if `nodeUpdate` works as expected? I think here that you'd need to unregister the old node object and register the new one if that passes (or vice versa). Otherwise we'd have a long list of nodes registered representing the same node object.

Reply:
A node update typically updates its status fields. Especially it is forbidden to change the providerID of the spec or its name. Therefore the node registration happens with the same values and the node manager is hopefully idempotent. I'm not an expert here, but removing the node registration on an update of its status looks to me like a very dangerous operation.

Reply:
The problem here is a timing behaviour. If the CSI controller is already running if new nodes join the cluster
the node object is created first without attributes required by the CS controller. They are later added by the cloud controller.
But this is then no create event anymore, but an update event. Therefore it is not possible to ignore update events in the CSI controller.